src/proposer/proposer.cpp

```diff
@@ -178,13 +219,26 @@ class ProposerImpl : public Proposer {
     m_waiter.Wake();
   }
 
+  std::shared_ptr<const CBlock> GenerateBlock(CWallet &wallet,
+                                              const staking::CoinSet &coins,
+                                              const boost::optional<CScript> &) override {
+
+    return proposer::GenerateBlock(*m_active_chain,
+                                   *m_transaction_picker,
+                                   *m_block_builder,
+                                   *m_proposer_logic,
+                                   wallet,
+                                   coins,
+                                   boost::none);
```

Comment:
nit: since `boost::none` is a value and not variable, would be nice to see a `/* coinbase_script */` comment.